tests/foreman/api/test_hostcollection.py

```diff
@@ -23,438 +23,464 @@
 from robottelo.api.utils import promote
 from robottelo.constants import DISTRO_RHEL7
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
-from robottelo.test import APITestCase
 from robottelo.vm import VirtualMachine
 
 
-class HostCollectionTestCase(APITestCase):
-    """Tests for host collections."""
-
-    @classmethod
-    def setUpClass(cls):
-        """Create hosts that can be shared by tests."""
-        super().setUpClass()
-        cls.org = entities.Organization().create()
-        cls.hosts = [entities.Host(organization=cls.org).create() for _ in range(2)]
-        cls.lce = entities.LifecycleEnvironment(organization=cls.org).create()
-        cls.content_view = entities.ContentView(organization=cls.org).create()
-        cls.content_view.publish()
-        cls.content_view = cls.content_view.read()
-        promote(cls.content_view.version[0], environment_id=cls.lce.id)
-        cls.activation_key = entities.ActivationKey(
-            content_view=cls.content_view, environment=cls.lce, organization=cls.org
-        ).create()
+@pytest.fixture(scope='module')
+def fake_hosts(module_org):
+    """Create content hosts that can be shared by tests."""
+    hosts = [entities.Host(organization=module_org).create() for _ in range(2)]
+    return hosts
+
+
+@pytest.fixture(scope='module')
+def content_view(module_org, module_lce, module_published_cv):
+    content_view = module_published_cv
```

Comment:
I think we can place this fixture in the api_fixtures file and replace return with yield and add the content_view cleanup.

Reply:
ACK, will test that

Reply:
Change made, now called `module_promoted_cv`